### PR TITLE
Integrate Method I in Method D

### LINKED ISSUE #808: Integrate Method I into Method D
With the introduction of the new parameter `use_blind_points` and `calc_IFD_only` we can compute method I in method D

- [x] Add mentioned parameters to Method D
- [x] Adopt tests of method I to run as Method D with corresponding parameters 
- [x] Check and adopt existing inifiles in repository

### PR DESCRIPTION
**Pull request by JetteSchumann**
Functionality of Method I is now available by configuring Method D with the parameters `use_blind_points` and `global_IFD`. For calculating global IFD data no measurement area needs to be defined anymore. A bounding box with the ID -1 is generated implicitly.
Unused code was removed and attributes were renamed for better understanding.

Still need to update tests and inifiles.
Since the default option for Method D changes (use of blind points is enabled by default) the tests need to be restructured.

Fixes #808

### CODE REVIEW

**Comment by JetteSchumann** [comment]
@gjaeger the results did not change. The reference tests checked whether the output has changed and they are are all successful (meaning: no changes). The tests have just been restructured since the default settings of Method D changed (`use_of_blind_points` enabled). To keep the old reference data of Method D, I had to create inififles with `use_blind_points` disabled. Therefore, the tests of `global_IFD`needed to be moved to a new set of tests since `use_blind_points` was enabled for the Method I.

File: systemtest/jpsreport_tests/reference_tests/blind_points_disabled/cut_off_edges/jpsreport_ini.xml [hunk 1]
```diff
@@ -57,20 +27,16 @@
...
   <method_D enabled="true">
-    <measurement_area id="5" start_frame="None" stop_frame="None" get_individual_FD="true"/>
-    <measurement_area id="8" start_frame="None" stop_frame="None" get_individual_FD="true"/>
+    <measurement_area id="5" start_frame="None" stop_frame="None" local_IFD="true"/>
+    <measurement_area id="8" start_frame="None" stop_frame="None" local_IFD="true"/>
     <one_dimensional enabled="false"/>
     <cut_by_circle enabled="true" radius="1.0" edges="20"/>
     <profiles enabled="false" grid_size_x="0.20" grid_size_y="0.20"/>
+    <use_blind_points enabled="false"/>
   </method_D>
```
[the remark this answers is not in the document]
Comment (JetteSchumann) [reply systemtest/jpsreport_tests/reference_tests/blind_points_disabled/cut_off_edges/jpsreport_ini.xml:37]: It was only used in Method I. This test was moved to /global_IFD/cut_off_edges/ (where no MA needs to be defined anymore).

File: systemtest/jpsreport_tests/method_D/D09_corridor_high_cut_off_no_blind_points/jpsreport_ini.xml [hunk 2]
```diff
@@ -20,7 +20,7 @@
 
     <method_D enabled="true">
         <measurement_area id="1" start_frame="None" stop_frame="None"
-                          get_individual_FD="true"/>
+                          local_IFD="true"/>
         <one_dimensional enabled="false"/>
         <cut_by_circle enabled="true" radius="1.0607" edges="100"/>
         <profiles enabled="false" grid_size_x="0.20" grid_size_y="0.20"/>
```
[the remark this answers is not in the document]
Comment (JetteSchumann) [reply systemtest/jpsreport_tests/method_D/D09_corridor_high_cut_off_no_blind_points/jpsreport_ini.xml:26]: I forgot that here. Thanks.

File: systemtest/jpsreport_tests/reference_tests/global_IFD/cut_off_large/test_cut_off_large.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+path.append(os.path.dirname(os.path.dirname(path[0])))
+from reference_test import check_diff_to_reference_data
+
+
+from utils import SUCCESS, FAILURE
+from JPSRunTest import JPSRunTestDriver
+
+def runtest(inifile, trajfile):
+
+    logging.info("=====  Test global IFD data of Method D with us_blind_points enabled, cut_by_circle enabled and various radius of 1.5, 2.0 and 2.5 ===============")
```
[the remark this answers is not in the document]
Comment (JetteSchumann) [reply systemtest/jpsreport_tests/reference_tests/global_IFD/cut_off_large/test_cut_off_large.py:23]: different radiuses have been tests for Method D, I and J. changed this

File: jpsreport/methods/Method_D.cpp [hunk 4]
```diff
@@ -93,53 +101,60 @@ bool Method_D::Process(
...
+
+        if(IdInFrame.size() == 0) {
+            LOG_WARNING("no pedestrians in frame <{}>", frameNr);
+            continue;
+        }
+
         //------------------------------Remove peds outside geometry------------------------------------------
-        for(int i = 0; i < (int) IdInFrame.size(); i++) {
-            if(false == within(point_2d(round(XInFrame[i]), round(YInFrame[i])), _geoPoly)) {
+        for(int i = 0; i < static_cast<int>(IdInFrame.size()); i++) {
```
[the remark this answers is not in the document]
Comment (JetteSchumann) [reply jpsreport/methods/Method_D.cpp:148]: done

File: jpsreport/methods/Method_D.cpp [hunk 5]
```diff
@@ -53,34 +53,42 @@ Method_D::Method_D()
...
 bool Method_D::Process(
     const ConfigData_DIJ & configData,
-    int measurementAreaID,
-    const PedData & peddata,
-    const fs::path & scriptsLocation,
+    int measurementAreaIndex,
+    const PedData & pedData,
     const double & zPos_measureArea)
 {
+    // TODO: lots of these parameters should be initialized in the constructor
```
Comment (zeroset) [inline jpsreport/methods/Method_D.cpp:67]: :+1:

File: jpsreport/methods/Method_D.cpp [hunk 6]
```diff
@@ -53,34 +53,42 @@ Method_D::Method_D()
...
+    auto _stopFrame    = configData.stopFrames[measurementAreaIndex];
+    auto _calcLocalIFD = configData.calcLocalIFD[measurementAreaIndex];
+
+    auto _getProfile        = configData.getProfile;
+    auto _isOneDimensional  = configData.isOneDimensional;
+    auto _useBlindPoints    = configData.useBlindPoints;
+    bool _calcGlobalIFDOnly = false;
+
+
+    if(_measurementArea->_id == -1) {
```
Comment (zeroset) [inline jpsreport/methods/Method_D.cpp:86]: We should get rid of this "special" MA with id -1.

**Comment by JetteSchumann** [comment]
@zeroset I renamed all Method D tests as we decided to put "_no_blind_points" (must be disabled explicitly) and "_global_IFD" (must be enabled explicitly) at the end of the tests.

The deleted Output file was accidentally added to the repo before.

**Pull request merged**